faculty.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'human'
+
+class Faculty < Human
+  attr_reader :age, :salary, :phase, :target_raise
+  attr_accessor :name
+
+  def initialize (options = {})
+
+  end
+
+  def salary=(new_salary)
+    puts "This better be good!"
+    @salary = new_salary
+  end
+
+  def receive_raise(raise)
+    @salary += raise
+  end
+
+  def teach_stuff
+   return self.class.to_s == "Teacher" ? "Listen, class, this is how everything works, fo SHO! *drops flat-out insane knowledge bomb* ... You're welcome. *saunters away*" : "Listen, class, this is how everything works. *drops crazy knowledge bomb* ... You're welcome."
```

Comment:
Nope nope nope, never check which subclass you're in from the superclass.  Just leave the methods in the subclasses and let inheritance do its thing.

If you really want to pull something up, pull the more default-seeming one up and only overwrite in the less default-y one.